airlock/views/code.py

```diff
@@ -0,0 +1,83 @@
+from django.http import Http404
+from django.shortcuts import redirect
+from django.template.response import TemplateResponse
+from django.views.decorators.clickjacking import xframe_options_sameorigin
+from django.views.decorators.http import require_http_methods
+from django.views.decorators.vary import vary_on_headers
+
+from airlock.business_logic import CodeRepo, bll
+from airlock.file_browser_api import get_code_tree
+from airlock.types import UrlPath
+from airlock.users import User
+from airlock.views.helpers import (
+    get_path_item_from_tree_or_404,
+    get_workspace_or_raise,
+    serve_file,
+)
+from services.tracing import instrument
+
+
+def get_repo_or_raise(user: User, workspace_name: str, commit: str):
+    workspace = get_workspace_or_raise(user, workspace_name)
+
+    try:
+        return CodeRepo.from_workspace(workspace, commit)
+    except bll.FileNotFound:
+        # cannot find manifest.json
+        raise Http404()
+    except (CodeRepo.RepoNotFound, CodeRepo.CommitNotFound):
+        raise Http404()
+
+
+# we return different content if it is a HTMX request.
+@vary_on_headers("HX-Request")
+@instrument(func_attributes={"workspace": "workspace_name", "commit": "commit"})
+def view(request, workspace_name: str, commit: str, path: str = ""):
+    repo = get_repo_or_raise(request.user, workspace_name, commit)
```

Comment:
Can/should we show a nicer error message to users if the repo or commit isn't found? Is it possible that a commit relsted to a particular is reverted and so we end up with a link to it that doesn't exist?

Reply:
Right, that's a good point. But it will depend on how we end up linking to this, which is a separate issue: #267 

I'll add a comment to that issue that we should handle that case. It may be that we have a specific page, or that we display a message that we couldn't find the commit.